pyrefly/lib/alt/call.rs

```diff
@@ -645,20 +648,36 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
                         metadata,
                     },
                 ),
-            ) => self.callable_infer(
-                signature,
-                Some(metadata.kind.as_func_id()),
-                tparams.as_deref(),
-                Some(obj),
-                args,
-                keywords,
-                range,
-                errors,
-                errors,
-                context,
-                hint,
-                ctor_targs,
-            ),
+            ) => {
+                let obj_for_helper = obj.clone();
+                let mut result = self.callable_infer(
+                    signature,
+                    Some(metadata.kind.as_func_id()),
+                    tparams.as_deref(),
+                    Some(obj),
+                    args,
+                    keywords,
+                    range,
+                    errors,
+                    errors,
+                    context,
+                    hint,
+                    ctor_targs,
+                );
+
+                // Only promote for user-defined methods
+                if let FunctionKind::Def(func_id) = &metadata.kind {
+                    result = self.promote_literal_string_method_return(
```

Comment:
At the risk of being pedantic, I think converting `str` -> `LiteralString` is technically a narrowing/demotion, since LiteralString is narrower than str.